src/software/ai/hl/stp/play/offense/offense_play_test.py

```diff
@@ -67,18 +69,21 @@ def test_offense_play(simulated_test_runner):
     )
 
     # Always Validation
-    # TODO- #2779 Validation
-    always_validation_sequence_set = [[]]
+    inv_always_validation_sequence_set = [
+        [BallAlwaysStaysInRegion(regions=[field.fieldBoundary()])]
+    ]
+
+    ag_always_validation_sequence_set = [[FriendlyAlwaysHasBallPossession()]]
 
     # Eventually Validation
-    # TODO- #2779 Validation
-    eventually_validation_sequence_set = [[]]
+    inv_eventually_validation_sequence_set = [[]]
+    ag_eventually_validation_sequence_set = [[FriendlyTeamEventuallyScored()]]
```

Comment:
@jameschuang2002 I added 2 aggregate tests checking that the friendly team to have possession and that friendly team eventually scored